src/lib/user-guiding.js

```diff
@@ -0,0 +1,124 @@
+const api = require('./api');
+const sample = require('lodash.sample');
+
+const USER_GUIDING_ID = process.env.USER_GUIDING_ID;
+const AUTONOMY_SURVEY_ID = 3048;
+const RELATIONSHIP_SURVEY_ID = 3049;
+const JOY_SURVEY_ID = 3050;
+const COMPETENCE_SURVEY_ID = 3045;
+const EDITOR_INTERACTION_SURVEY_IDS = [COMPETENCE_SURVEY_ID, JOY_SURVEY_ID];
+
+const CONDITIONS = {condition_list: [
+    'IsLoggedIn',
+    'NotAdmin',
+    'NotMuted'
+]};
+
+const USER_GUIDING_SNIPPET = `
+    (function(g, u, i, d, e, s) {
+        g[e] = g[e] || [];
+        var f = u.getElementsByTagName(i)[0];
+        var k = u.createElement(i);
+        k.async = true;
+        k.src = 'https://static.userguiding.com/media/user-guiding-' + s + '-embedded.js';
+        f.parentNode.insertBefore(k, f);
+        if (g[d]) return;
+        var ug = g[d] = {
+            q: []
+        };
+        ug.c = function(n) {
+            return function() {
+                ug.q.push([n, arguments])
+            };
+        };
+        var m = ['previewGuide', 'finishPreview', 'track', 'identify', 'hideChecklist', 'launchChecklist'];
+        for (var j = 0; j < m.length; j += 1) {
+            ug[m[j]] = ug.c(m[j]);
+        }
+    })(window, document, 'script', 'userGuiding', 'userGuidingLayer', '${USER_GUIDING_ID}');
+`;
+
+const activateUserGuiding = (userId, callback) => {
+    if (window.userGuiding) {
+        callback();
+        return;
+    }
+
+    const userGuidingScript = document.createElement('script');
+    userGuidingScript.innerHTML = USER_GUIDING_SNIPPET;
+    document.head.insertBefore(userGuidingScript, document.head.firstChild);
+
+    window.userGuidingSettings = {disablePageViewAutoCapture: true};
+
+    window.userGuidingLayer.push({
+        event: 'onload',
+        fn: () => window.userGuiding.identify(userId.toString())
+    });
+
+    window.userGuidingLayer.push({
+        event: 'onIdentificationComplete',
+        fn: callback
+    });
+};
+
+const attemptDisplayUserGuidingSurvey = (userId, permissions, guideId, callback) => {
+    if (!USER_GUIDING_ID || !process.env.SORTING_HAT_HOST) {
+        return;
+    }
+
+    api({
+        uri: '/user_guiding',
+        method: 'GET',
+        headers: {
+            'Content-Type': 'application/json',
```

Comment:
During Bug Hunt, I saw `text/json` in some places. From what I've read, `application/json` is more correct, so I don't recommend changing this, but it does remind me that we may have an inconsistency somewhere in the system.

CC @AlexisGoodfellow

Reply:
I dug into this a bit more and found this pretty insightful answer on SO: https://stackoverflow.com/a/69518701

According to IANA's page on media types (https://www.iana.org/assignments/media-types/media-types.xhtml) the `text/json` field is not an officially registered media type, so it might be more of a leftover from the original RFC's state.

Another detail to note here is that the `api` lib doesn't populate the `body` object that's passed in the callback if the content type of the response is `text/json`.